lib/logstash/inputs/kafka.rb

```diff
@@ -296,62 +298,99 @@ def kafka_consumers
     @runner_consumers
   end
 
-  private
+  def subscribe(consumer)
+    @pattern.nil? ? consumer.subscribe(topics) : consumer.subscribe(@pattern)
+    consumer
+  end
+
   def thread_runner(logstash_queue, consumer)
-    Thread.new do
+    java.lang.Thread.new do
```

Comment:
just curious, are there other reasons for not using the (Ruby) `Thread` here, besides `abort_on_exception` ?

this is now going to act as a Java adopted thread (in Ruby land) and thus the runtime "tear-down" mechanics would slightly diverge - I do not think it will matter much but there's a difference on the thread not being managed by JRuby.

Reply:
Yeah, `abort_on_exception` handling was my main concern here. 

I would prefer to leave the handling of uncaught exceptions to the [uncaught exception handler](https://github.com/elastic/logstash/blob/c698aa224b44c7356fbfeb58ebb0c4a66d72dc71/logstash-core/src/main/java/org/logstash/Logstash.java#L96), rather than deferring to `abort_on_exception` - `Thread.abort_on_exception = true` will crash logstash on any 'escaped' exception, `Thread.abort_on_exception = false` will fail to shutdown when encountering fatal errors, such as `java.lang.OutOfMemoryError`.

Reply:
oh right, have assumed things might work if `abort_on_exception` would be disabled per thread here.
but that still means raising on the "main" thread instead of propagating to the Java thread handler.
have basically assumed [this part](https://github.com/jruby/jruby/blob/9.2.16.0/core/src/main/java/org/jruby/RubyThread.java#L1796-L1798) would allow `java.lang.Error` instances to also propagate through.